integreat_cms/cms/models/regions/region.py

```diff
@@ -630,32 +630,73 @@ def get_language_or_404(self, language_slug, only_active=False, only_visible=Fal
         except KeyError as e:
             raise Http404("No language matches the given query.") from e
 
+    @cached_property
+    def explicitly_archived_ancestors_subquery(self):
+        """
+        This property returns a subquery for all explicitly archived anchestors of a given page.
+        Needs to be used as part of another query because in order to resolve :class:`~django.db.models.OuterRef`,
+        e.g. in a :class:`~django.db.models.Subquery` or in :class:`~django.db.models.Exists`.
+
+        This directly calls :meth:`~integreat_cms.cms.models.regions.region.Region.explicitly_archived_ancestors_query` with ``fixed=None``.
+
+        :return: A queryset of the explicitly archived ancestors.
+        :rtype: ~django.db.models.query.QuerySet
+        """
+        return self.explicitly_archived_ancestors_query(fixed=None)
+
+    def explicitly_archived_ancestors_query(self, fixed=None):
+        """
+        Returns a query for all explicitly archived ancestors of a given page.
+        If ``fixed`` is not None, returns a queryset for all explicitly archived ancestors of that particular page.
+        If ``fixed`` is None, returns a query containing :class:`~django.db.models.OuterRef`, such that it can be used
+        as part of another query e.g. in a :class:`~django.db.models.Subquery` or in :class:`~django.db.models.Exists`.
+
+        :param fixed: The page to get the explicitly archived ancestors for, or ``None``.
+        :type fixed: ~integreat_cms.cms.models.abstract_tree_node.AbstractTreeNode
+
+        :return: A queryset of the explicitly archived ancestors.
+        :rtype: ~django.db.models.query.QuerySet
+        """
+        if fixed:
+            tree = fixed.tree_id
+            lft = fixed.lft
+            rgt = fixed.rgt
+        else:
+            tree = models.OuterRef("tree_id")
+            lft = models.OuterRef("lft")
+            rgt = models.OuterRef("rgt")
+
+        return self.pages.filter(
+            tree_id=tree,
+            lft__lt=lft,
+            rgt__gt=rgt,
```

Comment:
```suggestion
        return self.pages.filter(
            tree_id=models.OuterRef("tree_id"),
            lft__lt=models.OuterRef("lft"),
            rgt__gt=models.OuterRef("rgt"),
```
I think one method for the subquery is enough here.